GitHub editing guide

rbot-github opened the pull request:
A guide on how to edit the BobaDocs code in github.

CHANGES:

- "Writing" section added to the Volunteer Guide
- A two page (guide and appendix) guide on how to edit BobaDocs, placed in the new writing section
- A new folder named `volunteer` made in `static/img`, along with a bunch of images used in the BobaDocs guide placed in that folder

essential-randomness commented on docs/volunteering/writing/editing-bobadocs.md:
```diff
@@ -0,0 +1,158 @@
+
+Doing so has led me to an overview of Codebase Overview. Near the top right of the file preview is a button shaped like a pencil. Hovering over it will give you its details: "Edit this file". Click it, and you'll be brought to GitHub's built-in editor.
+
+### Method 2: Go directly from the page you want to edit
```
Actually I think this is supposed to work without 404 and docusaurus is misconfigured. I can look into it if you want, or you can do that.

rbot-github replied:
Making a note here: fixing the Docusaurus configuration made it so clicking the edit button now leads directly to an editor that you can submit a pull request from. So that section needs to be reworked. I'll resolve this when it is.

essential-randomness commented on docs/volunteering/writing/editing-bobadocs.md:
```diff
@@ -0,0 +1,158 @@
+
+![a cropped screenshot of the bobadocs github page, with red arrows pointing to relevant spots on the branch dropdown](/img/volunteer/bobadocsbranch.jpg)
+
+## Step 1: Navigating to GitHub's editor
```
When it comes to steps, I would use an active voice: s/"Navigating"/"Navigate".

With that said, maybe it would be more like "navigate to the file to edit"? "find the page to edit"? I don't like either, but hopefully you get me.

Also, when a section has alternative methods, we should help people decide which method they prefer upfront (though in this case they're fairly equivalent). Personal preference, maybe, but I would also have the least complicated one as the first (to me this would be from the actual documentation website in the case of writers). 

 So in this case:

> ## Step 1: Find the right file
> 
> There are two ways to find the file corresponding to the page you want to edit: from the bobadocs website, or from within the github website. [[I really don't know what to say as far as the difference goes. For me they're somewhat equivalent, which can also be what we say.]]
> 
> ### Method 1 - Find file from BobaDocs Website
> 
> ### Method 2 - Find file from your fork

essential-randomness commented on docs/volunteering/writing/editing-bobadocs.md:
```diff
@@ -0,0 +1,158 @@
+
+If you look at your repo, you'll notice that it looks a lot like a filesystem. That's because it is a filesystem! And you can navigate to the different files in it - that is, the *pages in the site* - by clicking through it.
+
+To do that, we need to know a little bit about how Docusaurus is laid out. Docusaurus separates site content into four folders:
```
Part of me feels this should go in a different guide. Maybe you can say:

You can find files corresponding to documentation in the `docs` folder. You can read more about Docusaurus' structure [in our Docusaurus Knowledge Base].

essential-randomness commented on docs/volunteering/writing/editing-bobadocs.md:
```diff
@@ -0,0 +1,158 @@
+
+![a screenshot of the git repo, with a red arrow pointing to the docs folder](/img/volunteer/bobadocsnav1.jpg)
+
+You'll notice four new folders: `engineering`, `product`, `users`, and `volunteer`. Each of these folders corresponds to one of the guides on BobaDocs. Click the folder for the guide with the page you want to edit.
```
I would add a screenshot of the top bar, and do a guide here similar to what you did further up:

- `engineering`: [what goes here]
- `volunteers`: [what goes here]
...

rbot-github replied:
Top bar screenshot is added, but I'm disagreeing on the guide. A few of the requested changes (including the one for the other folder guide) are about moving out-of-scope information somewhere else, and I think explaining what each guide is fits into that. I handled the other one by moving it to a separate docusaurus guide, and linking to that in the section, and am doing the same for the guide about the guides.

If you insist that it should be in the github editing guide too, I'll move it back, but for the next pass/review I want to see if it's okay with that setup.

essential-randomness commented on docs/volunteering/writing/editing-bobadocs.md:
```diff
@@ -0,0 +1,158 @@
+
+You'll notice four new folders: `engineering`, `product`, `users`, and `volunteer`. Each of these folders corresponds to one of the guides on BobaDocs. Click the folder for the guide with the page you want to edit.
+
+Once there, you'll notice something a little odd:
```
> Once there, you'll notice that the names of files and folders are similar to the ones on the sidebar. This isn't a coincidence [etc...].
> [screenshot]

Make sure you give people full context on what they should be looking at in a screenshot.

essential-randomness commented on docs/volunteering/writing/editing-bobadocs.md:
```diff
@@ -0,0 +1,158 @@
+
+Some of the names of these things are QUITE SIMILAR. This isn't a coincidence! Docusaurus generates the sidebar for BobaDocs based on the files and folders present: each file is a page on the site, while each folder is a section on the sidebar.
+
+Armed with this information (and a little investigation), I determine that the page I want to edit, Codebase Overview, is named `intro.md` in the repo. I click it.
```
Did you mention beforehand which page you want to edit?

I would split these sections:

#### Understanding the repo structure

[previous explanation]

#### Finding your file

For the purpose of this guide, we'll edit the Codebase Overview page [link, potentially screenshot of the parts that are helpful].

You could point out that the structure of the folder is actually the same as the url:
> https://bobadocs.netlify.app/**docs/engineering/intro**

I would also point out the steps a bit more:

1 - Codebase Overview is not in a menu, so it has to be one of the files
2 - clicking around, I open intro.md and confirm that the title at the top of the page is the correct one

With that said, I should probably rename intro codebase-overview and make that a best practice TBH.

essential-randomness commented on docs/volunteering/writing/editing-bobadocs.md:
```diff
@@ -0,0 +1,158 @@
+
+Loading that URL, I now have GitHub's editor open, and can make my changes!
+
+## Step 2: Make your edits
```
I stopped here and haven't reviewed the rest. I'll pick it back up once you do a first pass :)

rbot-github commented:
I was hoping to get it done in one night, but I am being foolish. Leaving a note here so I can pick this back up later. Here's where I'm at so far:

DONE:
- Steps 1 and 3 have been edited to match requested changes
- Wrote technical git terms out of guide, replacing them with non-jargon equivalents
    - Information about branching has been removed from the main guide (moved to Knowledge Base). This includes one image, but I've saved a copy in case it's useful later
    - The exception to this is Step 4, which needs to be reworked (described below)
- Knowledge guide has been added to
    - Git Terminology section with a bunch of words defined and nothing else
    - Git FAQ section with two questions explaining why forking and branching are A Thing
- New Docusaurus Guide added to Engineering Guide
    - Information on how the folder structure for Docusaurus works moved there from Step 2 of guide
    - New information on what each guide covers is written there
- Suggestions on how to word things have been put in (where they're marked as resolved, at least

STILL TO DO:
- Steps 2 and 4 still need to be reworked/rewritten to reflect changes in the process
- Bullet poooooints
- Check that there's no bugs/it's not broken
- Final readthrough to check it all still makes sense
- Probably also check back through all edit requests to make sure I caught everything

Will finish this up tomorrow. For now, I'm going to bed

rbot-github commented:
Instead of rewriting steps 1 and 4, I ended up removing them entirely. The fixed "edit this page" links make it way easier to just go to the file directly, so I removed the route where you run around in github's filesystem. Similarly, since we're not asking them to branch, I moved the part where you clean up the branch (step 4), too. Now it's a note in the appendix in case anyone needs it.

I think mostly these edits cut out extra stuff, so hopefully it's more streamlined now. Give it a look when you can
